src/main/java/org/folio/circulation/item/service/CirculationItemsService.java

```diff
@@ -0,0 +1,85 @@
+package org.folio.circulation.item.service;
+
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.folio.circulation.item.domain.entity.Item;
+import org.folio.circulation.item.domain.mapper.CirculationItemsMapper;
+import org.folio.circulation.item.exception.IdMismatchException;
+import org.folio.circulation.item.exception.ResourceAlreadyExistException;
+import org.folio.circulation.item.repository.CirculationItemsRepository;
+import org.folio.rs.domain.dto.CirculationItem;
+import org.folio.spring.exception.NotFoundException;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class CirculationItemsService {
+
+  private final CirculationItemsRepository circulationItemsRepository;
+
+  private final CirculationItemsMapper circulationItemsMapper;
+
+  public CirculationItem getCirculationItemById(String id) {
+    var item = getCirculationItemEntityOrThrow(id);
+    return circulationItemsMapper.mapEntityToDto(item);
+  }
+
+  private Item getCirculationItemEntityOrThrow(String id) {
+    return circulationItemsRepository.findById(UUID.fromString(id))
+           .orElseThrow(() -> new NotFoundException(String.format("Circulation Item was not found by id= %s ", id)));
+  }
+
+
+    public CirculationItem createCirculationItem(String circulationItemId, CirculationItem circulationItem) {
+    if(Objects.nonNull(circulationItem.getId()) &&
+            !Objects.equals(circulationItemId, String.valueOf(circulationItem.getId()))) {
+      throw new IdMismatchException(
+              String.format("Request id= %s and entity id= %s are not equal", circulationItemId, circulationItem.getId())
+      );
+    }
+
+    if(Objects.isNull(circulationItem.getId())){
+      circulationItem.setId(UUID.fromString(circulationItemId));
+    }
+
+    if(checkIfItemExists(circulationItemId)){
+      throw new ResourceAlreadyExistException(
+        String.format("Unable to create circulation item with id %s as it already exists", circulationItemId));
+    }
+
+    var circulationItemEntity = circulationItemsMapper.mapDtoToEntity(circulationItem);
+    circulationItemEntity.setCreatedDate(LocalDateTime.now());
+
+    return circulationItemsMapper.mapEntityToDto(circulationItemsRepository.save(circulationItemEntity));
+  }
```

Comment:
i am not sure , but this code looks suspicious to me . 
what if some exception occurred while saving , what would be the output.
Please check . thanks

Reply:
I don't think, circulationItemsRepository.save should be wrapped with try/catch block.

please describe your suspicions in more detail

Reply:
i am not saying to add try and catch. i am just concern about the response that if any error occurred while saving then what would be the response .
Thanks

Reply:
the response is going to be according to the ExHandler of Exception.class. For such a case HttpStatus.INTERNAL_SERVER_ERROR